raiden/transfer/log.py

```diff
@@ -217,15 +239,26 @@ def log(self, state_change):
         serialized_data = self.serializer.serialize(state_change)
         return self.storage.write_state_change(serialized_data)
 
-    def log_events(self, state_change_id, events):
-        """ Log the events that were generated by `state_change_id into the write ahead Log
+    def log_events(self, state_change_id, events, current_block_number):
+        """ Log the events that were generated by `state_change_id` into the write ahead Log
         """
         assert isinstance(events, list)
         self.storage.write_state_events(
             state_change_id,
-            [(None, state_change_id, self.serializer.serialize(event)) for event in events]
+            [(None, state_change_id, current_block_number, self.serializer.serialize(event))
+             for event in events]
         )
 
+    def get_events_in_block_range(self, from_block, to_block):
+        """Get the raiden events in the period (inclusive) ranging from
+        `from_block` to `to_block`.
+
+        This function returns a list of tuples of the form:
+        (identifier, generated_statechange_id, block_number, event_object)
+        """
+        results = self.storage.get_events_in_range(from_block, to_block)
+        return [(res[0], res[1], res[2], self.serializer.deserialize(res[3])) for res in results]
```

Comment:
Please, use a `namedtuple` for the return value, these indices are getting quite confusing.

I didn't check all methods, but I guess this is useful for the state_changes too